src/plugins/i18n/index.js

```diff
@@ -0,0 +1,56 @@
+import {isEmpty} from 'lodash';
+
+export default class ChromeI18n {
+
+    constructor (translations = {}) {
+        this._translations = translations;
+    }
+
+    install (chrome) {
+        const plugin = this;
+        this.chrome = chrome;
+
+        Object.defineProperty(this.chrome, 'i18n', {
+            get: function () {
+                return plugin;
+            }
+        });
+    }
+
+    getMessage (messageName, ...substitutions) {
+        const {
+            message = undefined,
+            placeholders = {}
+        } = this._translations[messageName] || {};
+
+        if (isEmpty(substitutions) || isEmpty(placeholders)) {
+            return String(message);
+        }
+
+        function setPlaceholder(ignored, name) {
```

Comment:
Since it's a callback function to String.replace, I will refactor it to look more as such. Making it a private function might look a bit odd as its execution context will not be aligned with its purpose.